app/models/service_template.rb

```diff
@@ -69,6 +69,15 @@ class ServiceTemplate < ApplicationRecord
   scope :with_existent_service_template_catalog_id, ->         { where.not(:service_template_catalog_id => nil) }
   scope :displayed,                                 ->         { where(:display => true) }
 
+  def self.catalog_item_types
+    ci_types = Rbac.filtered(ExtManagementSystem.all).flat_map(&:supported_catalog_types).uniq
+    ci_types << 'generic_orchestration' if Rbac.filtered(OrchestrationTemplate).exists?
+    ci_types << 'generic'
```

Comment:
Since you are building this up as an index for lookup purposes (with uniqueness), you would be better off using a `Set` object (and not manually calling unique, but letting Set do that for you).